examples/energy-management-app/energy-management-common/include/ChargingTargetsMemMgr.h

```diff
@@ -0,0 +1,160 @@
+ *    needs to be called as each individual chargingTarget is loaded from persistent data.
+ *
+ *    Once the chargingTargets for the day schedule have been loaded, ChargingTargetsMemMgr::AllocAndCopy() is
+ *    called to allocate the memory to store the chargingTargets and the chargingTargets are copied.
+ *
+ * 2. When updating a Target and a day schedule is unaffected, the chargingTargets associated with
+ *    day schedule need copying. The following should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::List<const Structs::ChargingTargetStruct::Type> & chargingTargets)
+ *
+ * 3. When in SetTargets, a new list of chargingTargets needs to be added to a day schedule, the following
+ *    should be called:
+ *
+ *    ChargingTargetsMemMgr::AllocAndCopy(const DataModel::DecodableList<Structs::ChargingTargetStruct::DecodableType> &
+ * chargingTargets)
+ *
+ * Having allocated and copied the chargingTargets accordingly, they can be added to a
+ * DataModel::List<const Structs::ChargingTargetStruct::Type as follows:
+ *
+ * chargingTargetsList = DataModel::List<Structs::ChargingTargetStruct::Type>(ChargingTargetsMemMgr::GetChargingTargets(),
+ * ChargingTargetsMemMgr::GetNumDailyChargingTargets());
+ *
+ * All memory allocated by this object is released When the ChargingTargetsMemMgr destructor is called.
+ *
+ */
+
+class ChargingTargetsMemMgr
+{
+public:
+    ChargingTargetsMemMgr();
+    ~ChargingTargetsMemMgr();
+
+    /**
+     * @brief This method sets the index (mChargingTargetSchedulesIdx) to use into mpListOfDays.
+     *        This index is used in subsequent calls to GetChargingTargets and the AllocAndCopy
+     *        methods below.
+     *
+     *        NOTE: This method MUST be called each time a new day is added to
+     *        DataModel::List<const Structs::ChargingTargetScheduleStruct::Type> chargingTargetSchedules
+     *        before building up the ChargingTargetStruct associated with the mpListOfDays entry.
+     *
+     * @param chargingTargetSchedulesIdx  - The new index to use when accessing mpListOfDays
+     */
+    void Reset(uint16_t chargingTargetSchedulesIdx);
+
+    /**
+     * @brief Called as each individual chargingTarget is loaded from persistent data.
+     *        When loading the Target from persistent storage, it is not known upfront
+     *        how many chargingTargets are associated with this day schedule so
+     *        ChargingTargetsMemMgr::AddChargingTarget() needs to be called as each individual
+     *        chargingTarget is loaded from persistent data.
+     *
+     * @param chargingTarget  - The chargingTarget to add into mpListOfDays[mChargingTargetSchedulesIdx]
+     */
+    void AddChargingTarget(const EnergyEvse::Structs::ChargingTargetStruct::Type & chargingTarget);
+
+    /**
+     * @brief Called to allocate and copy the chargingTargets in mDailyChargingTargets to
+     *        mpListOfDays[mChargingTargetSchedulesIdx].
```

Comment:
reprhrase this as well. mDailyChargingTargets is a private field ... the caller does not know what this is.

The public API should be documented in a way that is understandable for use without referencing/re-stating the private implementation. I still do not know what AllocAndCopy intent is based on this description.

Reply:
what are `the chargingTargets` in the context of this call? I see no parameters here.

Reply:
I have reworded the API to avoid implementation details.